index.js

```diff
@@ -68,13 +77,25 @@ function commist (opts) {
     if (matching.length > 0) {
       matching[0].call(args)
 
-      // return null if there is nothing left to do
+      // return null to indicate there is nothing left to do
       return null
     }
 
     return args
   }
 
+  async function parseAsync (args) {
+    const matching = lookup(args)
+
+    if (matching.length > 0) {
+      await matching[0].call(args)
+      // return null to indicate there is nothing left to do
+      return Promise.resolve(null)
+    }
+
+    return Promise.reject(args)
```

Comment:
```suggestion
    return args
```

Reply:
We want to reject in this branch to be able to `.catch` on incorrectly parsed programs.

Reply:
Rejecting with a non-error is an antipattern. It's better to avoid it.

But this is not an error condition at all. Why should somebody use try/catch to catch a situation that's normal?

Reply:
Oh, how did I miss that? I'm even preaching this myself... https://twitter.com/kamilogorek/status/1596996078360203264 😅

Agree, let's keep it in sync with `parse` and resolve with `null` if successful and `args` if not matched.